airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -485,11 +486,9 @@ public KubePodProcess(final boolean isOrchestrator,
         .withLabels(labels)
         .withAnnotations(annotations)
         .endMetadata()
-        .withNewSpec();
-
-    if (isOrchestrator) {
```

Comment:
The reason for this flag is that we want elevated permissions for orchestrator pods. If we have a `CONNECTOR_JOB_KUBE_SERVICE_ACCOUNT_NAME` that defaults to null and an `ORCHESTRATOR_JOB_KUBE_SERVICE_ACCOUNT_NAME` that defaults to "airbyte-admin", then this section could be:

```
if (isOrchestrator) {
   podBuilder = podBuilder.withServiceAccount(orchestratorServiceAccount).withAutomountServiceAccountToken(true);
} else if (connectorServiceAccount != null) {
   podBuilder = podBuilder.withServiceAccount(connectorServiceAccount).withAutomountServiceAccountToken(true);
}
```

Then the default behavior would stay the same for both orchestrators and non-orchestrators, but users could choose to use any service account when creating either type of pod.

Reply:
Fixed in https://github.com/airbytehq/airbyte/pull/11697/commits/90ea13e8e51162a358a65154cf29508ec960e12b